src/librustc/lint/builtin.rs

```diff
@@ -2036,6 +2036,35 @@ impl LintPass for HardwiredLints {
     }
 }
 
+declare_lint! {
+    UNEXPORTED_NO_MANGLE,
```

Comment:
Thanks for the heads up. I didn't know that RFC existed. I have some mild
fud about priv purely because the keyword is gone but it might be confusing
to people who were around for that era (And I think they keyword is still
reserved, in which case this lint might end up being downright misleading).

As a strawman, how about `non-pub-no-mangle-fns` ? I realise that's a
mouthful, so I'd be totally ok with `private-no-mangle-fns` since it at
least doesn't conflict with a keyword. I tend to think that it should
include the name of the attr if at all possible to make grepping around for
things related to this do something reasonable.

On Thu, Jan 29, 2015 at 3:25 PM, Alex Crichton <notifications@github.com>
wrote:

> In src/librustc/lint/builtin.rs
> <https://github.com/rust-lang/rust/pull/21495#discussion_r23812524>:
>
> > @@ -2036,6 +2036,35 @@ impl LintPass for HardwiredLints {
> >      }
> >  }
> >
> > +declare_lint! {
> > +    UNEXPORTED_NO_MANGLE,
>
> In terms of a lint name, we may want to carefully consider our conventions
> <https://github.com/rust-lang/rfcs/blob/cc53afbe5dea41e1f7d1c3dce71e013abe025211/text/0344-conventions-galore.md>
> in this area. This may not be the best name as allow(unexported_no_mangle)
> may not quite fit those conventions.
>
> Some other possible names may be:
>
>    - private-no-mangle-fns
>    - priv-no-mangle-fns
>    - private-unmangled-fns
>
> Or some permutation of prefixes/suffixes. Our conventions say that because
> this only applies to functions it should probably have "fns" in the name
> somewhere though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/21495/files#r23812524>.
>